src/stdlib_experimental_stats_mean.fypp

```diff
@@ -12,28 +10,29 @@ submodule (stdlib_experimental_stats) stdlib_experimental_stats_mean
 
 contains
 
-  #:for k1, t1 in REAL_KINDS_TYPES
+  #:for k1, t1 in RC_KINDS_TYPES
     #:for rank in RANKS
-      module function mean_${rank}$_all_${k1}$_${k1}$(x, mask) result(res)
+      #:set RName = rname("mean_all",rank, t1, k1)
+      module function ${RName}$ (x, mask) result(res)
         ${t1}$, intent(in) :: x${ranksuffix(rank)}$
         logical, intent(in), optional :: mask
         ${t1}$ :: res
 
         if (.not.optval(mask, .true.)) then
-          res = ieee_value(res, ieee_quiet_nan)
+          res = ieee_value(real(res, kind=${k1}$), ieee_quiet_nan)
```

Comment:
`ieee_value` returns a NaN of the same type as `res`. So it should be OK without the conversion to a `real`, right? Or do I miss something?